zerver/lib/hotspots.py

```diff
@@ -75,12 +75,3 @@ def get_next_hotspots(user: UserProfile) -> List[Dict[str, object]]:
     user.tutorial_status = UserProfile.TUTORIAL_FINISHED
     user.save(update_fields=['tutorial_status'])
     return []
-
-def copy_hotpots(source_profile: UserProfile, target_profile: UserProfile) -> None:
-    for userhotspot in frozenset(UserHotspot.objects.filter(user=source_profile)):
-        UserHotspot.objects.create(user=target_profile, hotspot=userhotspot.hotspot,
-                                   timestamp=userhotspot.timestamp)
-
-    target_profile.tutorial_status = source_profile.tutorial_status
-    target_profile.onboarding_steps = source_profile.onboarding_steps
-    target_profile.save(update_fields=['tutorial_status', 'onboarding_steps'])
```

Comment:
All the changes in this commit should go along with the first commit.

The reason why we don't want a separate commit for removing `copy_hotpots` is that if we merge only the first commit, the system would be behaving incorrectly. That is `get_next_hotspots` would be fetching the hotspots from other profiles and `copy_hotpots` would be used used for copying the hotspots. We don't want that.  So both the changes should go in a single commit.

In general, whenever you create a commit x, ensure that the system works correctly if we deploy only the commits till x to production (even if the PR has many other commits after x).

Reply:
Thanks for the explanation, this makes sense.